oak/server/rust/oak_runtime/src/runtime/mod.rs

```diff
@@ -1186,22 +1186,26 @@ impl Runtime {
             .ok_or(OakStatus::ErrInvalidArgs)?;
 
         debug!("{:?}: start node instance {:?}", node_id, new_node_id);
-        self.node_start_instance(&new_node_name, instance, new_node_proxy, initial_handle)?;
+        let node_join_handle =
+            self.node_start_instance(&new_node_name, instance, new_node_proxy, initial_handle)?;
+
+        // Insert the now running instance to the list of running instances (by moving it), so that
+        // `Node::stop` will be called on it eventually.
+        self.add_node_join_handle(node_id, node_join_handle);
 
         Ok(())
     }
 
-    /// Starts running a newly created Node instance on a new thread.  The join handle
-    /// for the new thread is stored in the corresponding [`NodeInfo`].
+    /// Starts running a newly created Node instance on a new thread.
     /// The `node_name` parameter is only used for diagnostic/debugging output.
     fn node_start_instance(
-        self: Arc<Self>,
+        self: &Arc<Self>,
```

Comment:
It seems a bit odd to have a reference to an `Arc`, is this the only way it compiles?

Reply:
A normal `Arc` requires the callers to `.clone()` first, but it does allow the `.clone()` call inside the method to be dropped, so it's probably nicer this way around &ndash; done.